rust/private/rustc.bzl

```diff
@@ -290,6 +290,18 @@ def collect_inputs(
 
     linker_depset = cc_toolchain.all_files
 
+    # Pass linker additional inputs (e.g., linker scripts) only for linking-like
+    # actions, not for example where the output is rlib. This avoids quadratic
+    # behavior where transitive noncrates are flattened on each transitive
+    # rust_library dependency.
+    additional_transitive_inputs = []
+    if crate_info.type == "bin":
```

Comment:
I think you also want to flatten when crate type is "cdylib" and "dylib"

Reply:
Sounds good! Added both.